### PR TITLE
PLAT-71246 - Relationships guide [READY]

### PR DESCRIPTION
**Pull request by pnolan89**
https://jira.corp.adobe.com/browse/PLAT-71246

### CODE REVIEW

**Review by carlojavier — DISMISSED** [review]
LGTM!

File: api/guides/relationships.md [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+# Relationships
+
+Resources in Adobe Experience Platform Launch are often related to each other. This document provides an overview of how resource relationships are established in Platform Launch, and the relationship requirements of each resource type in the API.
+
+Depending on the type of resource in question, some relationships are required. A required relationship implies that the parent resource cannot exist without the relationship. All other relationships are optional.
+
+Regardless of whether they are required or optional, relationships are either automatically established by the system when relevant resources are created, or they must be created manually. In the case of creating relationships manually, there are two possible methods depending on the resource in question:
+
+* **Create by payload**: These relationships are created by providing a `relationship` object in the request payload when you create the parent resource. Examples of these relationships include:
+  * Create a `data_element` with the required `extensions`
```
Comment (thebenrob) [inline api/guides/relationships.md]: Can we link to the appropriate docs here: /api/reference/1.0/data_elements/create/

File: api/guides/relationships.md [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+  * Create an `environment` with the required `host` relationship.
```
Comment (thebenrob) [inline api/guides/relationships.md]: Link to /api/reference/1.0/environments/create/

File: api/guides/relationships.md [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+* **Create by URL**: These relationships are created by making a request to the `/relationships` endpoint of a particular resource, and supplying IDs and types in the request body for the resources you want to be related. This method only applies to libraries. Examples include:
+  * Creating a relationship between a library and an environment: `/libraries/{LIBRARY_ID}/relationships/environment`
```
Comment (thebenrob) [inline api/guides/relationships.md]: Link to http://localhost:8000/api/reference/1.0/libraries/relationships/environment/set_relationship/

File: api/guides/relationships.md [hunk 4]
```diff
@@ -0,0 +1,152 @@
...
+  * Creating a relationship between a library and a rule: `/libraries/{LIBRARY_ID}/relationships/rules`
```
Comment (thebenrob) [inline api/guides/relationships.md]: Link to http://localhost:8000/api/reference/1.0/libraries/relationships/rules/add_relationships/

**Review by thebenrob — CHANGES_REQUESTED** [review]
I think we should add some links to appropriate API specs in a few places.

The other thing I think we should do is add information about the format of the required payloads.  This could be either here in this guide (centralized in one place) or we could add it to the individual API spec pages that we link to - though if we go that route, we'll need to add the same info to other pages also.

For the Create by URL scenarios, I think we should say that inside the "data" object, you provide a "relationships" object.  That relationships object has another object that defines the relationship type you want to add ("extension" for example), then inside that a "data" object with an ID and a type.

It would also be worth specifying in whether you can add more than one a time.  Data elements for example only have one relationship, but Rule Components have two, and you have to specify both.

Create by URL - In these scenarios, you provide the "data" object, and then an array of objects inside it.  Each object needs to specify an "id" and a "type".  But you can add as many rules, data elements, and extensions as you want in a single call.

Now that I'm thinking about it, we should probably also say somewhere what the acceptable values of "type" are, which are: "data_elements", "rules", "extensions", and "environments".

**Review by thebenrob — DISMISSED** [review]
This is beautiful =)

**Review by thebenrob — APPROVED** [review]

**Pull request merged**